### PR TITLE
ci: add basic nix and cachix support

### PR DESCRIPTION
**Pull request by zimbatm**
pre-build the machines so they can be pulled from the cache

### CODE REVIEW

**Comment by ryantm** [comment]
I think we should modify the deploy script to use the subsituter and trusted-public-keys of nix-community.

**Comment by ryantm** [comment]
Also, does this mean we need to give our git-crypt secrets to GitHub?

**Comment by Mic92** [comment]
It should be actually possible to separate the secrets from the build process and not having the secrets in the nix store.

**Comment by zimbatm** [comment]
At the moment the `CACHIX_SIGNING_KEY` is added to the repo secrets and only available to GitHub Actions. To populate the cache you have to push a branch on the repo or get it merged into master.

File: deploy [hunk 1]
```diff
@@ -2,10 +2,17 @@
 #! nix-shell ./shell.nix -i bash
 set -euo pipefail
 
+options=(
+  --option extra-substituters "https://nix-community.cachix.org"
+  --option binary-cache-public-keys "cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs="
```
Comment (zimbatm) [inline deploy:7]: I don't know why but nixops seems to ignore those options

Comment (ryantm) [reply deploy:7]: Is your user trusted?

Comment (zimbatm) [reply deploy:7]: Yeah. Do you mind if the PR gets merged regardless? I don't think it hurts in its current state.

Comment (ryantm) [reply deploy:7]: I don't mind.

Comment (zimbatm) [reply deploy:7]: thanks! :)

**Pull request merged**